File: app/src/main/java/com/codepath/debuggingchallenges/activities/MainActivity.java
package com.codepath.debuggingchallenges.activities;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.codepath.debuggingchallenges.R;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    private void launchActivity(Class klass) {
        Intent intent = new Intent(this, klass);
        startActivity(intent);
    }

    public void launchCurrentDayActivity(View view) {
        launchActivity(CurrentDayActivity.class);
    }

    public void launchMoviesActivity(View view) {
        launchActivity(MoviesActivity.class);
    }

    public void launchChangeBackgroundActivity(View view) {
        launchActivity(ChangeBackgroundActivity.class);
    }

    public void launchToolbarActivity(View view) {
        launchActivity(ToolbarActivity.class);
    }

    public void launchSearchHistoryActivity(View view) {
        launchActivity(SearchHistoryActivity.class);
    }
}

File: app/src/main/java/com/codepath/debuggingchallenges/activities/MainActivity.kt
package com.codepath.debuggingchallenges.activities

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.codepath.debuggingchallenges.R
import android.content.Intent
import android.view.View
import com.codepath.debuggingchallenges.activities.CurrentDayActivity
import com.codepath.debuggingchallenges.activities.MoviesActivity
import com.codepath.debuggingchallenges.activities.ChangeBackgroundActivity
import com.codepath.debuggingchallenges.activities.ToolbarActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    private fun launchActivity(klass: Class<*>) {
        val intent = Intent(this, klass)
        startActivity(intent)
    }

    fun launchCurrentDayActivity(view: View?) {
        launchActivity(CurrentDayActivity::class.java)
    }

    fun launchMoviesActivity(view: View?) {
        launchActivity(MoviesActivity::class.java)
    }

    fun launchChangeBackgroundActivity(view: View?) {
        launchActivity(ChangeBackgroundActivity::class.java)
    }

    fun launchToolbarActivity(view: View?) {
        launchActivity(ToolbarActivity::class.java)
    }
}